File: src/main/java/com/durhack/sharpshot/nodes/routing/IfZeroNode.java
package com.durhack.sharpshot.nodes.routing;

import com.durhack.sharpshot.gui.Triangle;
import javafx.scene.Node;
import javafx.scene.paint.Color;
import org.jetbrains.annotations.NotNull;

public class IfZeroNode extends ConditionalNode {
    @Override
    public boolean branchingCondition(int signum) {
        return (signum == 0);
    }

    @Override
    public String toString() {
        return "Branch if Zero";
    }

    @Override
    public @NotNull Node toGraphic() {
        return new Triangle(getRotation(), Color.web("#FF0000"), "=0");
    }
}

File: src/main/java/com/durhack/sharpshot/nodes/routing/IfZeroNode.kt
package com.durhack.sharpshot.nodes.routing

import com.durhack.sharpshot.gui.shapes.Triangle
import javafx.scene.paint.Color
import java.math.BigInteger

class IfZeroNode : AbstractConditionalNode() {
    override fun branch(value: BigInteger?) = value?.signum() == 0
    override fun graphic() = Triangle(rotation, Color.web("#FF0000"), "=0")
    override val type = "branch if zero"
    override val tooltip = "Redirects all zero bullets (=0). Other bullets pass through unaffected"
}
